### PR TITLE
NMS-8267: Create documentation for SNMP detector with two examples

### PR DESCRIPTION
**Pull request by indigo423**
Create documentation for the SNMP detector with two short examples for table and scalar OID. Also created a template for new detectors to have similar structure.

* JIRA: http://issues.opennms.org/browse/NMS-8267
* Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS584

### CODE REVIEW

File: opennms-doc/guide-admin/src/asciidoc/text/provisioning/detectors/SnmpDetector.adoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+| `timeout`   | Timeout in milliseconds to wait for a response from the _SNMP_ agent.                                    | optional | _agent config_
+| `vbvalue`   | expected return value to detect the service; if not specified the service is detected if the _SNMP OID_
+                returned any kind of valid value.                                                                        | optional | `-`
+| `hex`       | Set `true` if the data is from type _HEX-String_.                                                        | optional | `false`
+| `isTable`   | Set `true` if detector should evaluate _SNMP tables_.                                                    | optional | `false`
+| `matchType` | Set match type to evaluate the expected value in the SNMP table. +
+                _EXIST_: the expected `vbalue` is ignored, service detected if the given table under _OID_ exist +
+                _ALL_: all values in the table must match against expected `vbalue` to detect service +
+                _ANY_: at least on value in the table must match against expected `vbalue` to detect service +
```
Comment (mvrueden) [inline opennms-doc/guide-admin/src/asciidoc/text/provisioning/detectors/SnmpDetector.adoc]: `on` vs. `one`

File: opennms-doc/guide-admin/src/asciidoc/text/provisioning/detectors/SnmpDetector.adoc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+===== Configuration and Usage
+
+.Parameters for the SNMP detector
+[options="header, autowidth"]
+|===
+| Parameter   | Description                                                                                              | Required | Default value
+| `oid`       | _SNMP OID_ for scalar or table to detect the service.                                                    | required | `.1.3.6.1.2.1.1.2.0`
+| `retry`     | Number of retries to detect the service.                                                                 | optional | _agent config_
+| `timeout`   | Timeout in milliseconds to wait for a response from the _SNMP_ agent.                                    | optional | _agent config_
+| `vbvalue`   | expected return value to detect the service; if not specified the service is detected if the _SNMP OID_
```
Comment (mvrueden) [inline opennms-doc/guide-admin/src/asciidoc/text/provisioning/detectors/SnmpDetector.adoc]: We should probably mention that reg expressions can be used.

**Comment by mvrueden** [comment]
Looks good to me. One side note though: In the example section the term `we` and `you` are used to directly address the reader. I am not sure if we do that anywhere else. We should probably stick with whatever we are doing everywhere else.

**Comment by indigo423** [comment]
👍  Thanks for reviewing and fixed findings.

**Pull request merged**